compiler/rustc_errors/src/emitter.rs

```diff
@@ -1930,6 +1939,24 @@ impl FileWithAnnotatedLines {
     }
 }
 
+fn replace_tabs(str: &str) -> String {
+    let mut str_vec: Vec<char> = str.chars().collect();
+    let mut tab_pos = vec![];
+    for (pos, c) in str_vec.iter().enumerate() {
+        if *c == '\t' {
+            tab_pos.push(pos);
+        }
+    }
+    // start with the tabs at the end of the line to replace them with 4 space chars
+    for pos in tab_pos.iter().rev() {
+        assert_eq!(str_vec.remove(*pos), '\t');
+        for _ in 0..4 {
+            str_vec.insert(*pos, ' ');
+        }
+    }
+    str_vec.into_iter().collect()
+}
```

Comment:
Why not just use `replace`?
```suggestion
fn replace_tabs(str: &str) -> String {
    s.replace('\t', " ")
}
```

Reply:
Ah, there is no reason, I happened to notice the existing tab replacement and didn’t think about std methods. Now that there’s no style to worry about, the std approach you suggest indeed seems simpler and clearer, so I’ll change to taht assuming the approach is okay. Thanks for the suggestion! 😄

Reply:
The previous logic was also modifying the style vector, and it _had_ to be done in this way so that the styles could also correspond to the new positions.